core/test/oracle/Voting.js

```diff
@@ -136,63 +158,26 @@ contract("Voting", function(accounts) {
     assert(await didContractThrow(voting.revealVote(identifier, time, newPrice, salt)));
     assert(await didContractThrow(voting.revealVote(identifier, time, price, newSalt)));
 
+    // Can't reveal with the incorrect address.
+    assert(await didContractThrow(voting.revealVote(identifier, time, newPrice, newSalt, { from: account2 })));
+
+    // Can't reveal with incorrect timestamp.
+    assert(await didContractThrow(voting.revealVote(identifier, (Number(time) + 1).toString(), newPrice, salt)));
+
+    // Can't reveal with incorrect identifier.
+    assert(
+      await didContractThrow(
+        voting.revealVote(web3.utils.utf8ToHex("wrong-identifier"), time, newPrice, newSalt, { from: account2 })
+      )
+    );
+
     // Successfully reveal the latest commit.
     await voting.revealVote(identifier, time, newPrice, newSalt);
 
     // Can't reveal the same commit again.
     assert(await didContractThrow(voting.revealVote(identifier, time, newPrice, newSalt)));
   });
 
-  it("Multiple voters", async function() {
```

Comment:
I deleted this test because it doesn't really add anything to the previous test. The "Multiple voters" test checks that different voters can't reveal each other's votes, but this is implied from the previous test which checks that the hashed `account` is correct.

Reply:
There's one other check in that test: that a voter who hasn't committed can't reveal. Are we testing that anywhere else?

Reply:
Isn't that the same thing as the test above? If a voter has not committed, then their `account` component of the hash must fail (revealVote checks `msg.sender`).

Reply:
I see, yeah, agreed.